core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -1290,6 +1291,14 @@ public static long longValue(RexNode node) {
     return number.longValue();
   }
 
+  /** Returns the value of a literal, cast, or unary minus, as a BigDecimal;
+   * never null. Uses {@link BigDecimal} for consistency with numeric literals
+   * and to facilitate precise comparison operations.*/
+  public static BigDecimal bigIntegerValue(RexNode node) {
```

Comment:
The function name should contain "round".
roundedValue perhaps?

Reply:
I think this name is more suitable than mine.